Muka-is-home/api
# Store Admin Functionality

This PR adds the ability to create, view, update and delete shop items from the admin portal as well as api access for frontend. Image hosting is done via Cloudinary, with admin able to upload files from their computer.

**To Test Admin:**
- add dependencies
```shell
pipenv install
```
- modify .env
  - I set up a temporary account with Cloudinary with my email, contact me for credentials or you can set up your own
  - add these variables to the .env:
    ```shell
    CLOUDINARY_CLOUD_NAME=
    CLOUDINARY_API_KEY=
    CLOUDINARY_API_SECRET=
    ```
- start server
```shell
python manage.py runserver
```
- log in as superuser and there should be a shop button in the navbar

**To test API:**
- Go to Postman and check these two endpoints:
  - http://localhost:8000/adminportal/shop
  - http://localhost:8000/adminportal/shop/[item_id]
